src/Security/ZitadelAuthenticator.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Security;
+
+use App\DTO\UserDto;
+use App\Entity\User;
+use App\Repository\UserRepository;
+use App\Service\IpResolver;
+use App\Service\UserManager;
+use App\Utils\Slugger;
+use Doctrine\ORM\EntityManagerInterface;
+use KnpU\OAuth2ClientBundle\Client\ClientRegistry;
+use KnpU\OAuth2ClientBundle\Security\Authenticator\OAuth2Authenticator;
+use thepaperpilot\OAuth2\Client\Provider\ZitadelResourceOwner;
```

Comment:
I sort of would've expected a composer vendor change for this, but I can't say for sure I know how either keycloak or zitadel work. Keycloak seems to pull in https://github.com/stevenmaguire/oauth2-keycloak/blob/master/src/Provider/KeycloakResourceOwner.php and adds it as a composer dependency

Reply:
Pushing a commit for PR feedback soon, but just wanted to point out that was actually the wrong path, from when I was still figuring stuff out. It'll be updated to App\Provider\ZitadelResourceOwner, as the file is within the repository itself.